test/jdk/java/util/zip/ZipFile/CenSizeTooLarge.java

```diff
@@ -35,55 +35,56 @@
 import java.io.BufferedOutputStream;
 import java.io.FileOutputStream;
 import java.io.IOException;
+import java.time.LocalDateTime;
 import java.util.zip.ZipEntry;
 import java.util.zip.ZipException;
 import java.util.zip.ZipFile;
 import java.util.zip.ZipOutputStream;
-import java.util.UUID;
 
-import static org.testng.Assert.assertThrows;
+import static org.testng.Assert.*;
 
-public class TestTooManyEntries {
-    // Number of directories in the zip file
-    private static final int DIR_COUNT = 25000;
-    // Number of entries per directory
-    private static final int ENTRIES_IN_DIR = 1000;
+public class CenSizeTooLarge {
+    // Maximum allowed CEN size allowed by the ZipFile implementation
+    private static int MAX_CEN_SIZE = Integer.MAX_VALUE - ZipFile.ENDHDR - 1;
 
     // Zip file to create for testing
     private File hugeZipFile;
 
     /**
-     * Create a zip file and add entries that exceed the CEN limit.
-     * @throws IOException if an error occurs creating the ZIP File
+     * Create a zip file with a CEN size which does not fit within a Java byte array
      */
     @BeforeTest
     public void setup() throws IOException {
-        hugeZipFile = File.createTempFile("hugeZip", ".zip", new File("."));
+        hugeZipFile = new File("cen-too-large.zip");
         hugeZipFile.deleteOnExit();
-        long startTime = System.currentTimeMillis();
+
         try (ZipOutputStream zip = new ZipOutputStream(new BufferedOutputStream(new FileOutputStream(hugeZipFile)))) {
-            for (int dirN = 0; dirN < DIR_COUNT; dirN++) {
-                String dirName = UUID.randomUUID() + "/";
-                for (int fileN = 0; fileN < ENTRIES_IN_DIR; fileN++) {
-                    ZipEntry entry = new ZipEntry(dirName + UUID.randomUUID());
-                    zip.putNextEntry(entry);
-                    zip.closeEntry(); // all files are empty
-                }
-                if ((dirN + 1) % 1000 == 0) {
-                    System.out.printf("%s / %s of entries written, file size is %sMb (%ss)%n",
-                            (dirN + 1) * ENTRIES_IN_DIR, DIR_COUNT * ENTRIES_IN_DIR, hugeZipFile.length() / 1024 / 1024,
-                            (System.currentTimeMillis() - startTime) / 1000);
-                }
+            // Creating the LocalDataTime once allows faster processing
+            LocalDateTime time = LocalDateTime.now();
+            long cenSize = 0;
+            // Add entries until MAX_CEN_SIZE is reached
+            for (int i = 0; cenSize < MAX_CEN_SIZE; i++) {
+                String name = Integer.toString(i);
```

Comment:
Consider making fewer but larger entries by generating large metadata.
Most obviously by making names in length close to the 64k metadata limit.
or by adding some fixed large extra field to each entry.

Reply:
Oh no, you reviewed just seconds before I pushed an update which does just that. I use a maximally sized extra field which makes the test run in 4 seconds using 12MB (measured using Runtime.freeMemory) of memory in the main loop.

Could I bother you to review the update version?